tests/parametric/test_span_links.py

```diff
@@ -175,28 +219,27 @@ def test_span_with_attached_links(self, test_agent, test_library):
                     second.add_link(first.span_id, attributes={"bools": [True, False], "nested": [1, 2]})
 
         traces = test_agent.wait_for_num_traces(1)
-        trace = find_trace_by_root(traces, Span(name="root"))
-        assert len(trace) == 3
+        assert len(traces[0]) == 3
 
-        root = root_span(trace)
+        root = traces[0][0]
         root_tid = root["meta"].get("_dd.p.tid") or "0" if "meta" in root else "0"
 
-        first = find_span(trace, Span(name="first"))
-        span = find_span(trace, Span(name="second"))
+        first = traces[0][1]
+        span = traces[0][2]
 
         assert span.get("parent_id") == root.get("span_id")
-        assert span.get("span_links") is not None
 
-        span_links = span["span_links"]
+        span_links = self._get_span_links(span)
         assert len(span_links) == 2
 
         link = span_links[0]
         assert link.get("span_id") == root.get("span_id")
         assert link.get("trace_id") == root.get("trace_id")
         assert (link.get("trace_id_high") or 0) == int(root_tid, 16)
         assert len(link.get("attributes") or {}) == 0
-        assert (link.get("tracestate") or "") == ""
-        assert (link.get("flags") or 0) == 0
+        # Ensure the tracestate from the parent span is set on the link
+        assert link.get("tracestate", "") == "dd=s:1;t.dm:-0"
+        assert link.get("flags", 0) == 1
```

Comment:
Where does this come from? I see no tracestate. The RFC explicitly calls out "This field must be omitted if no sampling information was provided".

Reply:
This tracestate and traceflags comes from the parent span's context ([python implementation](https://github.com/DataDog/dd-trace-py/blob/28b75850a72f670e5edc2bfa6c3935a7e2d30759/ddtrace/span.py#L546)). This is similar to the opentelemetry implementation. In python a span will always have at least two values in the tracestate (sampling decision and sampling priority). Does this align with the spec?

Reply:
I see. The PHP impl doesn't carry the tracestate/flags with individual spans, but keeps it as a property of the local trace root and we don't carry it around.

Would it make sense to leave it as implementation defined? I.e. assert that either tracestate and flags are properly set or not at all?